apps/serdes/halide_ir.fbs

```diff
@@ -0,0 +1,678 @@
+namespace Halide.Serialize;
+
+// This corresponds to the version.
+file_identifier "HLD0";
+
+// File extension of any written files. "hlpipe" stands for Halide Pipeline.
+file_extension "hlpipe";
+
```

Comment:
good point, this is a good approach to prevent IR update breaks serialization.

Reply:
The best we can do as far as I tried is to use the `file_identifier` field in serialization format, for example for Halide 17.0 we use "HL17"

Reply:
Right, but I think Derek is suggesting we add another struct with explicitly-populated metadata that may be useful / necessary for future consumers.

Reply:
oh I see, we can add that in the Pipeline like 
```
table Pipeline {
...
version: string;
}
```
with the string being something like "Halide v17.0.0-CHKSUM"?

Reply:
Yep. Though I suspect it might be better as a struct since we may well want to extend it, eg

table VersionMetadata {
   halide_version: string; // major.minor.patch
   halide_commit; string;  // git commit of Halide release
   halide_target: string;  // Target used for generation, or empty string if Target() used
}